### PR TITLE
Implement ngx.get_phase() FFI function

### PR DESCRIPTION
**Pull request by p0pr0ck5**
I hereby granted the copyright of the changes in this pull request
to the authors of this lua-nginx-module project.

### CODE REVIEW

**Comment by p0pr0ck5** [comment]
Sister PR at https://github.com/openresty/lua-resty-core/pull/78

File: src/ngx_http_lua_phase.c [hunk 1]
```diff
@@ -107,4 +107,21 @@ ngx_http_lua_inject_phase_api(lua_State *L)
     lua_setfield(L, -2, "get_phase");
 }
 
+#ifndef NGX_LUA_NO_FFI_API
```
Comment (membphis) [inline src/ngx_http_lua_phase.c]: style: need two blank line before this line

Comment (moonbingbing) [inline src/ngx_http_lua_phase.c]: committers will find lots of style problems, maybe we should add `ngx-releng`, `lua-releng`,`reindex` to travis CI, in order to save time.

Comment (p0pr0ck5) [inline src/ngx_http_lua_phase.c]: I would very much be in favor of this :)

File: src/ngx_http_lua_phase.c [hunk 2]
```diff
@@ -107,4 +107,22 @@ ngx_http_lua_inject_phase_api(lua_State *L)
...
+int
+ngx_http_lua_ffi_get_phase(ngx_http_request_t *r, char **err)
+{
+	ngx_http_lua_ctx_t  *ctx;
+
+	ctx = ngx_http_get_module_ctx(r, ngx_http_lua_module);
+	if (ctx == NULL) {
+		*err = "no request context";
+		return NGX_ERROR;
```
Comment (agentzh) [inline src/ngx_http_lua_phase.c]: Seems like bad indentation? We use 4-space indentation consistently in this module.

Comment (agentzh) [inline src/ngx_http_lua_phase.c]: And yeah, you should really use the `ngx-releng` tool to check your code's coding style to avoid common issues.

Comment (p0pr0ck5) [inline src/ngx_http_lua_phase.c]: My mistake :) I'll work on using ngx-releng and friends in a proper workflow (and set setexpandtab in vim ;) )

**Comment by p0pr0ck5** [comment]
Sister PR has been updated. Is there anything about this particular PR that needs to be addressed? I would be nice to get this in so Travis builds don't fail.

**Comment by agentzh** [comment]
@p0pr0ck5 You can always patch `.travis.yml` to point to your own branch of `lua-nginx-module` in your pull request's branch for `lua-resty-core` so that we can see whether travis ci passes on your branches before we merge anything.

**Comment by agentzh** [comment]
Fixed with the following commit log message:

```
feature: implemented pure C function for the pure FFI version of the ngx.get_phase() API function in lua-resty-core.
```

**Comment by agentzh** [comment]
Thanks!

**Pull request closed without merging**